### PR TITLE
quic: fixup tests

### PR DESCRIPTION
**Pull request by jasnell**
Fixing up some of the tests. There are still some failures here.

@addaleax: In particular, the fake-udp test is failing. Will be looking at that next week unless you happen to get to it first.

### CODE REVIEW

**Comment by jasnell** [comment]
Back the `once()`  changes back out

File: test/parallel/test-quic-process-cleanup.js [hunk 1]
```diff
@@ -21,19 +22,9 @@ const key = fixtures.readKey('agent1-key.pem', 'binary');
...
 
-server.listen({
-  key,
-  cert,
-  ca,
-  rejectUnauthorized: false,
-  maxCryptoBuffer: 4096,
-  alpn: 'meow'
-});
+server.listen({ key, cert, ca, alpn: 'meow' rejectUnauthorized: false });
```
Comment (addaleax) [inline test/parallel/test-quic-process-cleanup.js]:
```suggestion
server.listen({ key, cert, ca, alpn: 'meow', rejectUnauthorized: false });
```

File: test/parallel/test-quic-send-fd.js [hunk 2]
```diff
@@ -22,20 +23,10 @@ for (const variant of ['sendFD', 'sendFile', 'sendFD+fileHandle']) {
...
 
-  server.listen({
-    key,
-    cert,
-    ca,
-    rejectUnauthorized: false,
-    maxCryptoBuffer: 4096,
-    alpn: 'meow'
-  });
+  server.listen({ key, cert, ca, alpn: 'meow' rejectUnauthorized: false });
```
Comment (addaleax) [inline test/parallel/test-quic-send-fd.js]:
```suggestion
  server.listen({ key, cert, ca, alpn: 'meow', rejectUnauthorized: false });
```

File: test/parallel/test-quic-quicsocket.js [hunk 3]
```diff
@@ -112,34 +118,34 @@ socket.listen({ alpn: 'zzz' });
...
-  assert.strictEqual(typeof socket.address.family, 'string');
+  assert.strictEqual(typeof endpoint.address.address, 'string');
+  assert.strictEqual(typeof endpoint.address.port, 'number');
+  assert.strictEqual(typeof endpoint.address.family, 'string');
 
   // On Windows, fd will always be undefined.
   if (common.isWindows)
     assert.strictEqual(socket.fd, undefined);
   else
     assert.strictEqual(typeof socket.fd, 'number');
```
Comment (addaleax) [inline test/parallel/test-quic-quicsocket.js]: This assertion fails for me?

**Comment by addaleax** [comment]
> @addaleax: In particular, the fake-udp test is failing. Will be looking at that next week unless you happen to get to it first.

<details><summary>Diff (on top of this PR) in the fold</summary>

```diff
diff --git a/test/parallel/test-quic-qlog.js b/test/parallel/test-quic-qlog.js
index 9fef1d0853e7..748a7d9e2dba 100644
--- a/test/parallel/test-quic-qlog.js
+++ b/test/parallel/test-quic-qlog.js
@@ -18,7 +18,7 @@ const { serverSide, clientSide } = makeUDPPair();
 
 const server = quic.createSocket({
   validateAddress: true,
-  [kUDPHandleForTesting]: serverSide._handle,
+  endpoint: { [kUDPHandleForTesting]: serverSide._handle },
   qlog: true
 });
 
@@ -36,7 +36,7 @@ server.on('session', common.mustCall((session) => {
 
 server.on('ready', common.mustCall(() => {
   const client = quic.createSocket({
-    [kUDPHandleForTesting]: clientSide._handle,
+    endpoint: { [kUDPHandleForTesting]: clientSide._handle },
     client: { key, cert, ca, alpn: 'meow' },
     qlog: true
   });
@@ -44,7 +44,7 @@ server.on('ready', common.mustCall(() => {
 
   const req = client.connect({
     address: 'localhost',
-    port: server.address.port,
+    port: server.endpoints[0].address.port,
     qlog: true
   });
 
diff --git a/test/parallel/test-quic-with-fake-udp.js b/test/parallel/test-quic-with-fake-udp.js
index ba65984823b2..3a58729e2605 100644
--- a/test/parallel/test-quic-with-fake-udp.js
+++ b/test/parallel/test-quic-with-fake-udp.js
@@ -17,7 +17,7 @@ const ca = fixtures.readKey('ca1-cert.pem', 'binary');
 const { serverSide, clientSide } = makeUDPPair();
 
 const server = quic.createSocket({
-  [kUDPHandleForTesting]: serverSide._handle
+  endpoint: { [kUDPHandleForTesting]: serverSide._handle }
 });
 
 serverSide.afterBind();
@@ -38,14 +38,14 @@ server.on('session', common.mustCall((session) => {
 
 server.on('ready', common.mustCall(() => {
   const client = quic.createSocket({
-    [kUDPHandleForTesting]: clientSide._handle,
+    endpoint: { [kUDPHandleForTesting]: clientSide._handle },
     client: { key, cert, ca, alpn: 'meow' }
   });
   clientSide.afterBind();
 
   const req = client.connect({
     address: 'localhost',
-    port: server.address.port
+    port: server.endpoints[0].address.port
   });
 
   req.on('stream', common.mustCall((stream) => {
```

</details>

**Pull request merged**